lib/pure/osproc.nim

```diff
@@ -231,55 +230,79 @@ proc execProcesses*(cmds: openArray[string],
   ## executes the commands `cmds` in parallel. Creates `n` processes
   ## that execute in parallel. The highest return value of all processes
   ## is returned. Runs `beforeRunEvent` before running each command.
-  when false:
-    # poParentStreams causes problems on Posix, so we simply disable it:
-    var options = options - {poParentStreams}
-
+  
   assert n > 0
   if n > 1:
-    var q: seq[Process]
-    newSeq(q, n)
+    var i = 0
+    var q = newSeq[Process](n)
     var m = min(n, cmds.len)
-    for i in 0..m-1:
+
+    when defined(windows):
+      var w: WOHandleArray
+      var wcount = m
+
+    while i < m:
```

Comment:
Can you use ``for i in 0..<m`` here instead?

Reply:
This is answer on your question, compile it and run, please:
```nim
when isMainModule:
  var i = 0
  var m = 10
  for i in 0..<m:
    echo $i
  echo $i
```